doc/index.rst

```diff
@@ -9,6 +9,7 @@
 
    overview
    data_structures
+   dataset-conversions
```

Comment:
To follow the convention elsewhere: rename the file itself and here to `dataset_conversions`